python/dlr/counter/counter_mgr.py

```diff
@@ -0,0 +1,172 @@
+import platform
+import atexit
+import json
+import os
+import logging
+import threading
+
+from .publisher import MsgPublisher
+from .system import Factory
+from . import config
+from .model_exec_counter import ModelExecCounter
+from .model_metric import ModelMetric
+from .utils.helper import *
+
+
+def call_home(func):
+    def wrapped_call_home(*args, **kwargs):
+        call_counter = CallCounterMgr.get_instance()
+        if call_counter is not None:
+            if func.__name__ == "init_call_home":
+                print(config.CALL_HOME_USR_NOTIFICATION)
+                func(*args, **kwargs)
+                call_counter.runtime_loaded()
+            elif func.__name__ == "__init__":
+                func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_loaded(obj.get_model_name())
+            else:
+                res = func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_run(obj.get_model_name())
+                return res
+
+    return wrapped_call_home
+
+
+getinst = threading.Lock()
+relinst = threading.Lock()
+
+
+class CallCounterMgr(object):
+    RUNTIME_LOAD = 1
+    MODEL_LOAD = 2
+    MODEL_RUN = 3
+    _instance = None
+    DEFAULT_ENABLE_FEATURE = True
+
+    @staticmethod
+    def get_instance():
+        """return single instance of class"""
+        with getinst:
+            if CallCounterMgr._instance is None:
+                if CallCounterMgr.is_feature_enabled():
+                    try:
+                        CallCounterMgr._instance = CallCounterMgr()
+                    except Exception as e:
+                        CallCounterMgr._instance = None
+                        logging.exception("unsupported system for call home feature", exc_info=False)
+                    else:
+                        atexit.register(CallCounterMgr._instance.stop)
+                else:
+                    logging.warning("call home feature disabled")
+            return CallCounterMgr._instance
+
+    def __init__(self):
+        try:
+            machine_typ = platform.machine()
+            os_name = platform.system()
+            os_supt = "{0}_{1}".format(os_name, machine_typ)
+            self.system = Factory.get_system(os_supt)
+            if self.system is None:
+                #support all kind of linux distributions
+                if os_name.lower() == 'linux':
+                    self.system = Factory.get_system("Linux_ARM")
```

Comment:
I think we should expand support to linux os, rather than hard-coded machine architecture here.

Reply:
We have expanded support to Linux OS.